src/utils/sitemap_parser.py

```
@@ -0,0 +1,171 @@
+"""
+Sitemap parsing utilities for the MCP Crawl4AI RAG server.
+
+This module provides functionality to parse XML sitemaps, fetch sitemap URLs,
+and validate sitemap content. It extracts the sitemap parsing logic from
+crawling_tools.py into a focused, reusable module.
+"""
+
+import requests
+from typing import List, Optional, Union
```

Ruff (0.11.9)

`typing.List` is deprecated, use `list` instead

(UP035)